packages/armkit-cli/test/schema.test.ts

```diff
@@ -4,42 +4,119 @@ import * as path from 'path';
+    ["2019-04-01/Microsoft.VMwareCloudSimple", "VMwareCloudSimple"],
+    ["2018-10-15/Microsoft.LabServices", "LabServices"],
+    ["2018-04-01/Microsoft.Databricks", "Databricks"],
+    ["2020-09-01/Microsoft.StorageSync", "StorageSync"],
+    ["2019-06-01/Microsoft.HybridData", "HybridData"],
+    ["2021-05-01/Microsoft.Consumption", "Consumption"],
+    ["2020-06-01/Microsoft.EventGrid", "EventGrid"],
+    ["2021-03-01/Microsoft.StorageCache", "StorageCache"],
+    ["2018-03-01/Microsoft.Addons", "Addons"],
+    ["2019-09-01/Microsoft.Authorization", "Authorization"],
+    ["2021-05-01/Microsoft.Maintenance", "Maintenance"],
+    ["2019-12-01/Microsoft.SoftwarePlan", "SoftwarePlan"],
+    ["2020-09-01/Microsoft.Subscription", "Subscription"],
+    ["2018-05-01/Microsoft.SerialConsole", "SerialConsole"],
+    ["2021-04-01/Microsoft.Management", "Management"],
+    ["2017-04-01/Microsoft.Aadiam", "Aadiam"],
+    ["2021-01-01/Microsoft.PolicyInsights", "PolicyInsights"],
+    ["2020-06-01/Microsoft.AppConfiguration", "AppConfiguration"],
+    ["2016-01-29/Microsoft.PowerBI", "PowerBI"],
+    ["2018-09-15/Microsoft.DevTestLab", "DevTestLab"],
+    ["2020-01-01/Microsoft.DBforMySQL", "DBforMySQL"],
+    ["2020-05-01/Microsoft.SignalRService", "SignalRService"],
+    ["2021-03-01/Microsoft.Datadog", "Datadog"],
+    ["2020-07-07/Microsoft.OffAzure", "OffAzure"],
+    ["2021-04-01/Microsoft.AlertsManagement", "AlertsManagement"],
+    ["2021-03-01/Microsoft.DataBox", "DataBox"],
+  ];
 
-  await importer.import({outdir: workdir, targetLanguage: Language.TYPESCRIPT})
-
-  const output = fs.readFileSync(path.join(workdir, `ContainerInstance.ts`), 'utf-8');
-  expect(output).toMatchSnapshot();
-});
-
-test('container_registry_schema.test', async () => {
-  const workdir = fs.mkdtempSync(path.join(os.tmpdir(), 'container_registry.test'));
-  const importer = new ImportArmSchema(`["2019-05-01/Microsoft.ContainerRegistry"]`);
-
-  await importer.import({outdir: workdir, targetLanguage: Language.TYPESCRIPT})
-
-  const output = fs.readFileSync(path.join(workdir, `ContainerRegistry.ts`), 'utf-8');
-  expect(output).toMatchSnapshot();
-});
-
-test('container_service_schema.test', async () => {
-  const workdir = fs.mkdtempSync(path.join(os.tmpdir(), 'container_service.test'));
-  const importer = new ImportArmSchema(`["2020-09-01/Microsoft.ContainerService"]`);
-
-  await importer.import({outdir: workdir, targetLanguage: Language.TYPESCRIPT})
-
-  const output = fs.readFileSync(path.join(workdir, `ContainerService.ts`), 'utf-8');
-  expect(output).toMatchSnapshot();
-});
-
-test('storage_schema.test', async () => {
-  const workdir = fs.mkdtempSync(path.join(os.tmpdir(), 'storage.test'));
-  const importer = new ImportArmSchema(`["2019-06-01/Microsoft.Storage"]`);
+test.each(services)('%s.test', async (schemaConfig, serviceName) => {
+  const workdir = fs.mkdtempSync(path.join(os.tmpdir(), serviceName + '.test'));
```

Comment:
I think it would be more clear having be seperate test  function used by the loop

Reply:
What would go into the test function, as distinct from what's in the test function? Do you mean like this?

```

test.each(services)('%s.test', async (schemaConfig, serviceName) => {
  const workdir = fs.mkdtempSync(path.join(os.tmpdir(), serviceName + '.test'));
  const importer = new ImportArmSchema(`["` + schemaConfig + `"]`);
  importerTest(workdir, serviceName);
});

```
Or: 

```
test.each(services)('%s.test', importerTest(schemaConfig, serviceName);
```

Reply:
after discussion, we'll go for the latter option